lib/client/renderer.js

```diff
@@ -472,14 +487,130 @@ function init (client, d3) {
+            'text-anchor': 'middle',
+            dy: '.35em',
+            transform: 'rotate(-90 ' + (deleteRect.x + deleteRect.width / 2) + ',' + (deleteRect.y + deleteRect.height / 2) + ')'
+          })
+          .text(translate('Remove'));
+
+        if (treatment.insulin && treatment.carbs) {
+          carbsRect = {
+            x: 0,
+            y: 0,
+            width: chart().charts.attr('width'),
+            height: 50
+          };
+          insulinRect = {
+            x: 0,
+            y: chart().yScale(chart().yScale.domain()[0]) - 50,
+            width: chart().charts.attr('width'),
+            height: 50
+          };
+          chart().drag.append('rect')
+            .attr({
+              class:'drag-droparea',
+              x: carbsRect.x,
+              y: carbsRect.y,
+              width: carbsRect.width,
+              height: carbsRect.height,
+              fill: 'white',
+              opacity: 0.4,
+              rx: 10,
+              ry: 10
+            });
+          chart().drag.append('text')
+            .attr({
+              class:'drag-droparea',
+              x: carbsRect.x + carbsRect.width / 2,
+              y: carbsRect.y + carbsRect.height / 2,
+              'font-size': 15,
+              'font-weight': 'bold',
+              fill: 'white',
+              'text-anchor': 'middle',
+              dy: '.35em'
+            })
+            .text(translate('Move carbs'));
+          chart().drag.append('rect')
+            .attr({
+              class:'drag-droparea',
+              x: insulinRect.x,
+              y: insulinRect.y,
+              width: insulinRect.width,
+              height: insulinRect.height,
+              fill: '#0099ff',
+              opacity: 0.4,
+              rx: 10,
+              ry: 10
+            });
+          chart().drag.append('text')
+            .attr({
+              class:'drag-droparea',
+              x: insulinRect.x + insulinRect.width / 2,
+              y: insulinRect.y + insulinRect.height / 2,
+              'font-size': 15,
+              'font-weight': 'bold',
+              fill: '#0099ff',
+              'text-anchor': 'middle',
+              dy: '.35em'
+            })
+            .text(translate('Move insulin'));
+        }
+          
+        chart().basals.attr('display','none');
+          
+        operation = 'Move';
       })
       .on('drag', function() { 
         //console.log(d3.event);
         client.tooltip.transition().style('opacity', .9);
-        newTime = new Date(chart().xScale.invert(d3.event.x));
+        var x = Math.min(Math.max(0, d3.event.x), chart().charts.attr('width'));
+        var y = Math.min(Math.max(0, d3.event.y), chart().focusHeight);
+        
+        operation = 'Move';
+        if (isInRect(x, y, deleteRect) && isInRect(x, y, insulinRect)) {
+          operation = "Remove insulin"
+        } else if (isInRect(x, y, deleteRect) && isInRect(x, y, carbsRect)) {
+          operation = "Remove carbs"
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1901559821/issues/source?bid=2452379&fileBranchId=3084282#l600)